src/Source.jl

```diff
@@ -131,14 +133,15 @@ function Source(;fullpath::Union{AbstractString,IO}="",
         columntypes = types
     elseif isa(types, Dict) || isempty(types)
         columntypes = Vector{Type}(cols)
+        categoricals = Dict{Int, Set{String}}(i=>Set{String}() for i = 1:cols)
```

Comment:
Wouldn't using a `WeakRefString` be OK here to avoid some allocations?

Reply:
done